Sources/Segment/Utilities/Networking/HTTPSession.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+
+#if os(Linux) || os(Windows)
+import FoundationNetworking
+#endif
+
+public protocol DataTask {
+  var state: URLSessionTask.State { get }
+  func resume()
+}
+
+// An enumeration of default `HTTPSession` configurations to be used
+// This can be extended buy consumer to easily refer back to their configured session.
+public enum DefaultHTTPSession {
+  static let urlSession: HTTPSession = {
+      let configuration = URLSessionConfiguration.ephemeral
+      configuration.httpMaximumConnectionsPerHost = 2
+      let session = URLSession(configuration: configuration, delegate: nil, delegateQueue: nil)
+      return session
+  }()
+}
+
+public protocol HTTPSession {
+  func uploadTask(with request: URLRequest, fromFile file: URL, completionHandler: @escaping @Sendable (Data?, URLResponse?, Error?) -> Void) -> DataTask
```

Comment:
This didn't seem to work for me since in `URLSession` these functions return either a `URLSessionDataTask` or a `URLSessionUploadTask` and upload task inherits from data task, so using one associatedtyped confused the compiler since it didn't know how to map the returning type.

```
C:\Users\brian\Development\bcny-analytics-swift\Sources\Segment\Utilities\Networking\HTTPSession+Apple.swift:10:1: error: type 'URLSession' does not conform to protocol 'HTTPSession'
extension URLSession: HTTPSession {
^
C:\Users\brian\Development\bcny-analytics-swift\Sources\Segment\Utilities\Networking\HTTPSession.swift:25:18: note: ambiguous inference of associated type 'DataTaskType': 'URLSessionDataTask' vs. 'URLSessionUploadTask'
  associatedtype DataTaskType: DataTask
                 ^
FoundationNetworking.URLSession:19:15: note: matching requirement 'dataTask(with:completionHandler:)' to this declaration inferred associated type to 'URLSessionDataTask'
    open func dataTask(with request: URLRequest, completionHandler: @escaping (Data?, URLResponse?, (any Error)?) -> Void) -> URLSessionDataTask
              ^
FoundationNetworking.URLSession:24:15: note: matching requirement 'uploadTask(with:fromFile:completionHandler:)' to this declaration inferred associated type to 'URLSessionUploadTask'
    open func uploadTask(with request: URLRequest, fromFile fileURL: URL, completionHandler: @escaping (Data?, URLResponse?, (any Error)?) -> Void) -> URLSessionUploadTask
              ^
error: fatalError
```

I think I could solve it if you could have two `associatedtype`s on a protocol, but it doesn't look like you can do that sadly :(